src/lib/utilities/is-promise.test.ts

```diff
@@ -0,0 +1,79 @@
+import { isPromise } from './is-promise';
+
+describe(isPromise, () => {
+  it('should return true if an object is a resolved promise', async () => {
+    expect(isPromise(Promise.resolve())).toBe(true);
+  });
+
+  it('should return true if an object is a rejected promise', async () => {
+    try {
+      expect(isPromise(await Promise.reject().catch())).toBe(true);
+    } catch (er) {}
+  });
+
+  it('should return true if an object is a promise', () => {
+    expect(isPromise(new Promise(() => {}))).toBe(true);
+  });
+
+  it('should return false if an object is a function', () => {
+    expect(isPromise(() => {})).toBe(false);
+  });
+
+  it('should return false if an object is partially promise like with a .then', () => {
+    expect(isPromise({ then: () => {} })).toBe(false);
+  });
+
+  it('should return false if an object is partially promise like with a .catch', () => {
+    expect(isPromise({ catch: () => {} })).toBe(false);
+  });
+
+  it('should return false if an object is partially promise like with a .finally', () => {
+    expect(isPromise({ finally: () => {} })).toBe(false);
+  });
+
+  it('should return false if an object is promise like (contains .then, .catch. finally)', () => {
```

Comment:
Why? If it adheres to the contract, do we care?

Reply:
I'm happy to remove the `obj.toString() === [object Promise]` because that's what the last check is. I'm happy to have it be a promise if it has those methods.